modules/hostap/src/supp_events.c

```diff
@@ -9,10 +9,164 @@
 #include "includes.h"
 #include "common.h"
 
-int send_wifi_mgmt_event(const char *ifname, enum net_event_wifi_cmd event, int status)
+#define MAC_STR_FORMAT "%2hhx:%2hhx:%2hhx:%2hhx:%2hhx:%2hhx"
+
+static char *wpa_supp_event_map[] = {
+	"CTRL-EVENT-CONNECTED",
+	"CTRL-EVENT-DISCONNECTED",
+	"CTRL-EVENT-ASSOC-REJECT",
+	"CTRL-EVENT-AUTH-REJECT",
+	"CTRL-EVENT-TERMINATING",
+	"CTRL-EVENT-SSID-TEMP-DISABLED",
+	"CTRL-EVENT-SSID-REENABLED",
+	"CTRL-EVENT-SCAN-STARTED",
+	"CTRL-EVENT-SCAN-RESULTS",
+	"CTRL-EVENT-SCAN-FAILED",
+	"CTRL-EVENT-BSS-ADDED",
+	"CTRL-EVENT-BSS-REMOVED",
+	"CTRL-EVENT-NETWORK-NOT-FOUND",
+	"CTRL-EVENT-NETWORK-ADDED",
+	"CTRL-EVENT-NETWORK-REMOVED",
+};
+
+static int process_status(struct supp_status_data *event_data, char *status)
+{
+	int ret = 1; /* For cases where parsing is not being done*/
+	int event = -1;
+	int i;
+	unsigned char *mac;
+	union supp_mon_event_data *data;
+
+	data = (union supp_mon_event_data *)event_data->data;
+
+	for (i = 0; i < ARRAY_SIZE(wpa_supp_event_map); i++) {
+		if (strncmp(status, wpa_supp_event_map[i], strlen(wpa_supp_event_map[i])) == 0) {
+			event = i;
+			break;
+		}
+	}
+
+	if (i >= ARRAY_SIZE(wpa_supp_event_map)) {
+		return -ENOTSUP;
+	}
+
+	event_data->event = event + NET_EVENT_WPA_SUPP_CMD_COUNT + 1;
+
+	switch (event_data->event) {
+	case WPA_SUPP_CONNECTED:
+		mac = data->connected.bssid;
+		ret = sscanf(status+strlen("CTRL-EVENT-CONNECTED - Connection to"),
+			MAC_STR_FORMAT, &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
+		if (ret <= 0) {
```

Comment:
I guess this can be moved outside switch case, no?

Reply:
Done